consensus/clique/src/test/java/tech/pegasys/pantheon/consensus/clique/jsonrpc/methods/CliqueGetSignerMetricsTest.java

```diff
@@ -0,0 +1,257 @@
+
+import tech.pegasys.pantheon.consensus.clique.CliqueBlockInterface;
+import tech.pegasys.pantheon.consensus.clique.TestHelpers;
+import tech.pegasys.pantheon.crypto.SECP256K1;
+import tech.pegasys.pantheon.ethereum.core.Address;
+import tech.pegasys.pantheon.ethereum.core.BlockHeader;
+import tech.pegasys.pantheon.ethereum.core.BlockHeaderTestFixture;
+import tech.pegasys.pantheon.ethereum.core.Hash;
+import tech.pegasys.pantheon.ethereum.core.Util;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.JsonRpcRequest;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.exception.InvalidJsonRpcParameters;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.parameters.JsonRpcParameter;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.queries.BlockchainQueries;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcSuccessResponse;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.results.SignerMetricResult;
+import tech.pegasys.pantheon.util.uint.UInt256;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.LongStream;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class CliqueGetSignerMetricsTest {
+
+  private final JsonRpcParameter jsonRpcParameter = new JsonRpcParameter();
+  private final String CLIQUE_METHOD = "clique_getSignerMetrics";
+  private final String JSON_RPC_VERSION = "2.0";
+  private CliqueGetSignerMetrics method;
+
+  private BlockchainQueries blockchainQueries;
+  private CliqueBlockInterface cliqueBlockInterface;
+
+  @Rule public ExpectedException expectedException = ExpectedException.none();
+
+  @Before
+  public void setup() {
+    blockchainQueries = mock(BlockchainQueries.class);
+    cliqueBlockInterface = mock(CliqueBlockInterface.class);
+    method = new CliqueGetSignerMetrics(cliqueBlockInterface, blockchainQueries, jsonRpcParameter);
+  }
+
+  @Test
+  public void returnsCorrectMethodName() {
+    assertThat(method.getName()).isEqualTo(CLIQUE_METHOD);
+  }
+
+  @Test
+  public void exceptionWhenInvalidStartBlockSupplied() {
+    final JsonRpcRequest request = requestWithParams("INVALID");
+
+    expectedException.expect(InvalidJsonRpcParameters.class);
+    expectedException.expectMessage("Invalid json rpc parameter at index 0");
+
+    method.response(request);
+  }
+
+  @Test
+  public void exceptionWhenInvalidEndBlockSupplied() {
+    final JsonRpcRequest request = requestWithParams("1", "INVALID");
+
+    expectedException.expect(InvalidJsonRpcParameters.class);
+    expectedException.expectMessage("Invalid json rpc parameter at index 1");
+
+    method.response(request);
+  }
+
+  @Test
+  public void getSignerMetricsWhenNoParams() {
+
+    Long startBlock = 1L;
+    Long endBlock = 3L;
+
+    when(blockchainQueries.headBlockNumber()).thenReturn(endBlock);
+
+    List<SignerMetricResult> signerMetricResultList = new java.util.ArrayList<>();
```

Comment:
nit: Shouldn't don't need the full package name here
```suggestion
    List<SignerMetricResult> signerMetricResultList = new ArrayList<>();
```

Reply:
Also for the other `new java.util.ArrayList()` calls below.

Reply:
yes an error that I had not seen. I correct that